_sources/Unit5-Writing-Classes/topic-5-1-parts-of-class.rst

```diff
@@ -37,49 +37,131 @@
+``Turtle`` and we used instances of Java’s ``String`` class to assign values to
+different ``String`` variables.
 
-Watch this short video to review the vocabulary in object-oriented programming:
+Watch this short video to review the vocabulary of object-oriented programming:
 
 .. youtube:: LfSaSANJPLg
     :width: 650
     :height: 415
     :align: center
 
-You can think of a class as like a blueprint or a cookie cutter or a factory that produces objects. For example, the class House below is a blueprint of a house and the objects which are individual houses built from that blueprint. You can't live on a blueprint although it describes a house! But you can use it to build houses and live in those.
 
-To write your own class, you typically start a class declaration with ``public`` then ``class`` then the name of the class.  The body of the class is defined inside a ``{`` and a ``}``. For example, the class House below. Then, you can create objects of that new House type by using ``Classname objectname = new Classname();``
+Creating a Class
+------------------
+
+Most classes you write will start with the keyword ``public`` though it is not
+always required. But the class definition proper always starts with the word
+``class`` followed by the name of the class. Then the rest of the class, called
+the body, is defined inside a pair of ``{}``\ s.
+
+Since we’re talking about anatomy, let’s create a class called ``Person``.
+Classes are almost always named with capitalized names though this is a matter
+of style, not a rule of the language. Here is the basic skeleton of a ``Person``
+class:
 
 .. code-block:: java
 
-    public class House
+    public class Person
     {
-       // define class here - a blueprint
-
+       // define class here - also called the “body” of the class
     }
 
-    House myHouse = new House();
-    House neighborsHouse = new House();
-
-
-Remember that objects have attributes and behaviors. These correspond to **instance variables** and **methods** in the class definition.   Instance variables hold the data for objects where as the methods code the behaviors or the actions the object can do.   A class also has **constructors** which initialize the instance variables when the object is created, for example new House() above.  And a class can also have a **main method** which can be used to test the class.
+You can create instances of the ``Person`` class with ``new`` as in ``new
+Person()`` And you can declare variables that can hold a reference to a
+``Person`` object with ``Person variableName``.
 
-Let's create a class called Person. What would we want to know about a person?  What we want to know depends on what problem we are trying to solve.  In one situation, perhaps when creating an address book, we might want to know the person's name and phone number and email. Here's a possible Person class with instance variables, constructor, and methods.
-
-.. figure:: Figures/personClassDefn.png
```

Comment:
I think this figure got deleted somehow, although you still refer to it with the pink box.

Reply:
I don't think so. The pink box is in:

```
.. figure:: Figures/PersonClassWithCookies.png
```

which is still there. I think this figure was the picture of the `Person` code which I took out because the code is right there and the picture was really hard to read.